cocotb/handle.py

```diff
@@ -656,13 +656,16 @@ def _set_value(self, value, call_sim):
         """
         value, set_action = self._check_for_set_action(value)
 
-        if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
-            call_sim(self._handle.set_signal_val_long, set_action, value)
+        if isinstance(value, int):
+            if value >= -0x80000000 and value <= 0x7fffffff and len(self) <= 32:
```

Comment:
```suggestion
            if len(self) <= 32 and -(2**31) <= value <= (2**31)-1:
```

Reply:
Nice, didn’t know the combined comparison syntax was supported